src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -463,6 +463,20 @@ struct AddSub<T1, T2, ArgsList1<Args1...>, ArgsList2<Args2...>, Sign>
     }
   }
 
+  /// \brief Get a component from a `Tensor` in this expression's subtree of the
+  /// RHS `TensorExpression`
+  ///
+  /// \return a component from a `Tensor` in this expression's subtree of the
+  /// RHS `TensorExpression`
+  SPECTRE_ALWAYS_INLINE auto get_used_for_size() const {
```

Comment:
Ok, my only hangup with the alternative suggestions is that the function returns a component (e.g. `DataVector`), not a size (e.g. `5`), so I wouldn't want it to be called `get_component_size` when it returns a component, not the size. Do you have a preference for (1) using the name `get_component_size` and then I update the function to return the size instead of a component or (2) using a name like `get_rhs_tensor_component` and keeping the return type as-is?

Reply:
Oh yea, why not return a size?

Reply:
Ok, I'll do that! I assume then the return type should be `const auto` instead of `const auto&` since it will be an integral type?

Reply:
just do `size_t`. `Utilities/ContainerHelpers.hpp` has a `get_size` function that you can call